ttg/ttg/make_tt.h

```diff
@@ -44,15 +46,32 @@ class CallableWrapTT
 
   template <typename Key, typename Tuple>
   void call_func(Key &&key, Tuple &&args, output_terminalsT &out) {
-    func(std::forward<Key>(key), std::forward<Tuple>(args), out);
+    if constexpr (funcT_receives_outterm_tuple)
+      func(std::forward<Key>(key), std::forward<Tuple>(args), out);
+    else {
+      this->set_outputs_tls_ptr();
```

Comment:
Should this be moved into the backend TT (or some sort of wrapper) so that it can be used by applications with custom TT classes?